vertical-pod-autoscaler/docs/sidecar-containers.md

```diff
@@ -0,0 +1,173 @@
+# VPA Sidecar Container Management
+
+The Vertical Pod Autoscaler (VPA) has specific behavior when dealing with sidecar containers that are injected into pods via admission webhooks. This document explains the default behavior and how to customize it for your needs.
+
+## Understanding VPA and Container Policies
+
+### Default Container Policies
+
+To understand why sidecar handling is important, let's first look at how VPA manages containers by default. This default behavior is at the root of why special consideration is needed for sidecars.
+
+When you create a VPA resource for a pod, it automatically attempts to manage ALL containers in that pod - not just the ones you explicitly configure. This happens because:
+
+1. VPA applies a default `containerPolicy` with `mode: Auto` to any container not explicitly configured
+2. This automatic inclusion means VPA will try to manage resources for every container it sees
+3. Even sidecars injected into the pod will fall under VPA's management unless special steps are taken
+
+This default "manage everything" approach can cause problems with sidecars because:
+- Sidecar containers often have their own resource requirements set by their injection webhooks
+- VPA's automatic management may conflict with these requirements
+- Without proper handling, this can lead to resource conflicts and pod instability
+
+Example of a VPA resource that explicitly configures one container but affects all:
+
+```yaml
+apiVersion: autoscaling.k8s.io/v1
+kind: VerticalPodAutoscaler
+metadata:
+  name: my-app-vpa
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: my-app
+  updatePolicy:
+    updateMode: "Auto"
+  resourcePolicy:
+    containerPolicies:
+    - containerName: main-container
+      minAllowed:
+        cpu: 100m
+        memory: 50Mi
+# Note: Other containers will still get the default Auto mode
+```
+
+## Default Behavior: Ignoring Sidecar Containers
+
+### The vpaObservedContainers Annotation
+
+VPA uses a special annotation to track which containers were present in the pod before any webhook injections:
+
+```yaml
+metadata:
+  annotations:
+    vpaObservedContainers: "main-container,logging-sidecar"
+```
```

Comment:
This snippet is part of the Pod Kind, but the docs aren't clear about that:
```suggestion
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    vpaObservedContainers: "main-container,logging-sidecar"
```
```

Reply:
The snippet suggestion I did was broken, due to all the backtickets, I'll try again